### PR TITLE
Add support for Ubuntu 18.04 PostgreSQL 10

### PR DESCRIPTION
**Pull request by manics**
- Adds support for the upstream PostgreSQL on Ubuntu 18.04
- Removes support for 9.5
- Splits the role into two, a client role: https://github.com/manics/ansible-role-postgresql-client and this one (which depends on the client role). See https://github.com/ome/ansible-example-omero-onenode/pull/7#discussion_r387286874
- Reorganises the tests. Since the role is simplfied by removing some options and dropping 9.5 it is no longer necessary to split the tests into so many files, previously this was necessary due to differing behaviour between versions of PostgreSQL.

Other than dropping 9.5 the breaking changes shouldn't matter too much, as mentioned on https://github.com/ome/ansible-example-omero-onenode/pull/7#discussion_r387286874 it seems the expectation is that the role installs the server anyway, and the installation of extensions doesn't have any functional effect as they they're not automatically activated.

This is required to manage OMERO.server on Ubuntu 18.04: https://github.com/ome/ansible-role-omero-server/pull/50

Tag: new feature, `5.0.0`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,18 +5,18 @@ Postgresql
 [![Ansible Role](https://img.shields.io/ansible/role/41080.svg)](https://galaxy.ansible.com/ome/postgresql/)
 
 Install upstream PostgreSQL server.
+
 Optionally creates users and databases.
+If you wish to use your distributions packages do not use this role.
```
Comment (mtbc) [inline README.md]:
```suggestion
If you wish to use your distribution's packages then do not use this role.
```

File: molecule/centos7/tests/utils.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ver_lookup = {
+    "postgresql-96-localhost": "9.6",
+    "postgresql-10-localhost": "10",
+    "postgresql-11-localhost": "11",
+    "postgresql-92-localhost": "12",
```
Comment (mtbc) [inline molecule/centos7/tests/utils.py]: not "postgresql-12-localhost"?
(also below)

**Comment by manics** [comment]
I've added @mtbc's suggestions, and also combined the tests for Centos-7 and Ubuntu-1804 into one. The advantage is less duplication of tests, the disadvantage is more logic in the tests to take account of differences in the distributions.

**Comment by manics** [comment]
@jburel I've also initiated a transfer of the dependent repo https://github.com/manics/ansible-role-postgresql-client to you

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,12 @@
+# Changes in Version 5
+
+## Summary of breaking changes
+
+- PostgreSQL 9.5 is no longer supported by this role.
+- `postgresql_install_server` is removed, the server is always configured, use `ome.postgresql_client` to install just the client.
```
Comment (joshmoore) [inline CHANGES.md:6]: :+1:

**Comment by jburel** [comment]
repo transferred to the org

File: CHANGES.md (hunk 3 above)
Comment (joshmoore) [reply CHANGES.md:6]: Is it worth failing if it's set to `False`?

File: molecule/resources/requirements.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+- name: ome.postgresql_client
+  src:
+    https://github.com/manics/ansible-role-postgresql-client/archive/10d6b6858ec3614f83301040b5c9f29a3360da29.tar.gz
```
Comment (jburel) [inline molecule/resources/requirements.yml]: ``ansible-role-postgresql-client`` has now been migrated to the org.

File: CHANGES.md (hunk 3 above)
Comment (manics) [reply CHANGES.md:6]: Done in https://github.com/ome/ansible-role-postgresql/pull/19/commits/73bb1101803f7129c4149bd55602ef686b0d3d06

File: molecule/resources/requirements.yml (hunk 4 above)
Comment (manics) [reply molecule/resources/requirements.yml]: Can you tag it so it goes to galaxy (0.1.0?)

Comment (manics) [reply molecule/resources/requirements.yml]: Travis isn't activated https://travis-ci.org/ome/ansible-role-postgresql-client
so the tag wasn't pushed to Galaxy 😞

Comment (jburel) [reply molecule/resources/requirements.yml]: it should be sorted now
travis is building master and the tag

**Comment by jburel** [comment]
since we will have to add support for ubuntu 20.04, using "ubuntu" will not work

**Comment by manics** [comment]
Can you be more specific please? Is this in relation to a particular file?

File: molecule/resources/tests/utils.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Enumerate instead of parsing so we pick up errors if a host is added/changed
+ver_lookup = {
+    "postgresql-96-c7": ("9.6", "centos"),
+    "postgresql-10-c7": ("10", "centos"),
+    "postgresql-11-c7": ("11", "centos"),
+
+    "postgresql-96-u1804": ("9.6", "ubuntu"),
```
Comment (jburel) [inline molecule/resources/tests/utils.py:7]: this lookup will need to be reviewed when we add support for centos 8 and ubuntu 20.04

**Comment by jburel** [comment]
Added

File: molecule/resources/tests/utils.py (hunk 5 above)
Comment (manics) [reply molecule/resources/tests/utils.py:7]: It might not be necessary if the postgres package has the same layout on disk across distro versions

Comment (jburel) [reply molecule/resources/tests/utils.py:7]: let see when the time comes

**Comment by jburel** [comment]
Due to the change in this PR, we need to go with the suggested tag i.e. ``5.0.0``

**Review by jburel — APPROVED** [review]

**Comment by sbesson** [comment]
Note the `5.0.0` tag has not been imported into https://galaxy.ansible.com/ome/postgresql as per the recent Galaxy issues - see https://github.com/ome/ansible-role-omero-web/pull/36#issuecomment-621675469. You might need to reimport the role manually.
These recurring import issues have been reported upstream - see https://github.com/ansible/galaxy/issues/2362 and https://github.com/ansible/galaxy/issues/2369 so hopefully a fix will come up soon.

**Comment by manics** [comment]
Imported:
```
git checkout 5.0.0
ansible-galaxy login --github-token <SECRET-TOKEN>
ansible-galaxy import ome ansible-role-postgresql
```
So potentially this could be done in Travis.

**Pull request merged**